src/main/capi/arrow-c.cpp

```diff
@@ -355,12 +355,6 @@ void EmptyStreamRelease(ArrowArrayStream *stream) {
 
 void FactoryGetSchema(ArrowArrayStream *stream, ArrowSchema &schema) {
 	stream->get_schema(stream, &schema);
-
-	// Need to nullify the root schema's release function here, because streams don't allow us to set the release
-	// function. For the schema's children, we nullify the release functions in `duckdb_arrow_scan`, so we don't need to
-	// handle them again here. We set this to nullptr and not EmptySchemaRelease to prevent ArrowSchemaWrapper's
-	// destructor from destroying the schema (it's the caller's responsibility).
-	schema.release = nullptr;
```

Comment:
So: this is a copy, and `PopulateArrowTableSchema` looks like it expects ownership of the copy.